src/app/components/overlay/ConfirmDependentOverlay.jsx

```diff
@@ -1,57 +1,177 @@
+    : count === 0
+    ? "table:dependent-rows.no-dependent-rows-header"
+    : "table:dependent-rows.dependent-rows-header";
+  return i18n.t(trnKey, { count });
+};
 
-const RowsOverlay = props => {
-  const [depMessage, setDepMessage] = useState(
-    <p>{i18n.t("table:fetching_dependent_rows")}</p>
+const DeletionInfoBox = ({ table, row, nLinks, onSubmit, langtag }) => {
+  const headline = getHeadline(true, nLinks);
+  const textKey = `table:dependent-rows.delete-${
+    nLinks === 0 ? "no-" : ""
+  }dependent-rows`;
+
+  const handleSetMergeRowAction = mergeRowId =>
+    onSubmit(MergeRows(row.id, mergeRowId));
+  const handleSetJustDeleteAction = () => onSubmit(JustDelete(row.id));
+  const handleSelectMergeRowId = () => {
+    openSelectLinkTargetOverlay({
+      row,
+      table,
+      langtag,
+      onSubmit: handleSetMergeRowAction
+    });
+  };
+
+  return (
+    <>
+      <div className="deletion-info__header overlay-subheader__title">
+        {headline}
+      </div>
+      <div className="deletion-info__message overlay-subheader__description">
+        {i18n.t(textKey)}
+      </div>
+      {nLinks > 0 ? (
+        <div className="deletion-info__action-select overlay-subheader__buttons overlay-subheader__buttons--left">
+          <Button classNames="negative" onClick={handleSetJustDeleteAction}>
+            {i18n.t("table:dependent-rows.btn-select-delete-row")}
+          </Button>
+          <Button onClick={handleSelectMergeRowId}>
+            {i18n.t("table:dependent-rows.btn-select-merge-rows")}
+          </Button>
+        </div>
+      ) : null}
+    </>
   );
+};
 
-  const { table, row, langtag, deleteInfo, grudData, cell } = props;
+const DeletionFooter = ({ deletionAction, onClose, langtag, tableId }) => {
+  const deleteTextKey =
+    deletionAction && isMergeAction(deletionAction)
+      ? "table:dependent-rows.btn-submit-merge-rows"
+      : "table:dependent-rows.btn-submit-delete-row";
+
+  const handleSubmit = () =>
+    handleDeleteRow({ langtag, tableId, deletionAction });
+
+  const isSubmitDisabled =
+    f.isNil(deletionAction) ||
+    (isMergeAction(deletionAction) &&
+      f.isNil(deletionAction.mergedLinkTargetId));
+
+  return (
+    <footer className="button-wrapper">
+      <div className="action-buttons">
+        <Button
+          classNames="negative"
+          disabled={isSubmitDisabled}
+          onClick={handleSubmit}
+        >
+          {i18n.t(deleteTextKey)}
+        </Button>
+        <Button className="neutral" onClick={onClose}>
+          {i18n.t("common:cancel")}
+        </Button>
+      </div>
+    </footer>
+  );
+};
 
-  const hasDependencies = () =>
-    setDepMessage(<p>{i18n.t("table:delete_row_dependent_text")}</p>);
+const handleDeleteRow = ({ tableId, langtag, deletionAction }) => {
+  const rowId = deletionAction.rowToDeleteId;
+  const mergeWithRowId = deletionAction.mergedLinkTargetId;
+  const rows = store.getState() |> f.get(["rows", tableId, "data"]);
```

Comment:
**[question]** Der Pipeline operator ist ja bisher nur ein proposal und noch nicht final. Aktuell ist er in stage 2 (draft). Sollten wir ihn bereits nutzen? Es könnte sich in Zukunft nochmal ändern, wie er genutzt werden sollte.

Reply:
Das hatten wir vor ein paar Jahren mal diskutiert und den dann eingeführt, weil er ja "schon fast durch" war. Leider hat sich seitdem nichts mehr getan. Prinzipiell könnte man den natürlich aus dem ganzen Projekt wieder rausfaktorisieren, würde ich aber wenn dann mal als eigenen Task machen. Der Code hier ist übrigens nichts neues, nur verschoben.